src/Interpreters/Access/InterpreterCreateSettingsProfileQuery.cpp

```diff
@@ -48,16 +48,21 @@ BlockIO InterpreterCreateSettingsProfileQuery::execute()
     else
         getContext()->checkAccess(AccessType::CREATE_SETTINGS_PROFILE);
 
+    std::optional<SettingsProfileElements> settings_from_query;
+    if (query.settings)
+    {
+        settings_from_query = SettingsProfileElements{*query.settings, access_control};
+
+        if (!query.attach)
```

Comment:
I guess it can be checked earlier

Reply:
We actually use `settings_from_query` further in the code regardless of `query.attach`, so it seems like checking it earlier would not make sense